module/zfs/zvol.c

```diff
@@ -821,14 +817,14 @@ zvol_request(struct request_queue *q, struct bio *bio)
 		 * Some requests are just for flush and nothing else.
 		 */
 		if (uio.uio_resid == 0) {
-			if (bio->bi_rw & VDEV_REQ_FLUSH)
+			if (bio_is_flush(bio))
```

Comment:
In order having to restructure this code for multiple kernel I extended the `bio_is_flush()` check to cover both REQ_PREFLUSH and REQ_OP_FLUSH in newer kernels.  This isn't always 100% optimal since we might do an extra flush but it should be functionally correct and keep the code simpler.  We can always optimize this if its an issue.